packages/context/src/signalr/SignalRConnectionProvider.tsx

```diff
@@ -34,143 +33,57 @@ export const SignalRConnectionContext = createContext<SignalRConnectionContextTy
-                            if (documentsConnectionInfo.accessToken) {
-                                return documentsConnectionInfo.accessToken;
-                            } else {
-                                return getConnection();
-                            }
-                        },
-                    };
-
-                    const connection = new HubConnectionBuilder()
-                        .withUrl(documentsConnectionInfo.url, options as unknown as HttpTransportType)
-                        .configureLogging(LogLevel.Error)
-                        .withAutomaticReconnect()
-                        .build();
-
-                    setDocumentsSignalRConnection(connection);
-                }
-                // eslint-disable-next-line no-empty
-            } catch (err) {}
-        };
-
-        getConnection();
-    }, []);
-
-    useEffect(() => {
-        let documentsConnectionStopped = false;
-
-        const startConnection = async (connection: HubConnection, numOfAttempts: number) => {
-            await connection.start().catch((error: Error) => {
-                if (numOfAttempts < 4 && !documentsConnectionStopped) {
-                    setTimeout(() => {
-                        if (!documentsConnectionStopped) {
-                            startConnection(connection, numOfAttempts + 1);
-                        }
-                    }, 2000);
-                } else {
-                    console.warn(`Cannot start connection to SignalR due to error "${error}"`);
-                }
-            });
-        };
-
-        if (documentsSignalRConnection) {
-            startConnection(documentsSignalRConnection, 0);
-        }
-
-        return () => {
-            documentsSignalRConnection?.stop();
-            documentsConnectionStopped = true;
-        };
-    }, [documentsSignalRConnection]);
-
-    useEffect(() => {
-        let instancesConnectionStopped = false;
-
-        const startConnection = async (connection: HubConnection, numOfAttempts: number) => {
-            await connection.start().catch((error: Error) => {
-                if (numOfAttempts < 4 && !instancesConnectionStopped) {
-                    setTimeout(() => {
-                        if (!instancesConnectionStopped) {
-                            startConnection(connection, numOfAttempts + 1);
-                        }
-                    }, 2000);
-                } else {
-                    console.warn(`Cannot start connection to SignalR due to error "${error}"`);
-                }
-            });
-        };
-
-        if (instancesSignalRConnection) {
-            startConnection(instancesSignalRConnection, 0);
-        }
-
-        return () => {
-            instancesSignalRConnection?.stop();
-            instancesConnectionStopped = true;
-        };
-    }, [instancesSignalRConnection]);
+    const notifications = useNotification();
+
+    const [instanceCallbacks] = useState(
```

Comment:
@actong I approved it cuz this would work, but do we need useState here, or can we just put this in a constant?

Reply:
If it was just a local const, it would get reset every time the component re-renders.  Or you mean it'd be a const at the module level?